packages/@uppy/golden-retriever/src/index.js

```diff
@@ -113,8 +113,10 @@ export default class GoldenRetriever extends BasePlugin {
 
     // If all files have been removed by the user, clear recovery state
     if (Object.keys(filesToSave).length === 0) {
-      this.uppy.setState({ recoveredState: null })
-      MetaDataStore.cleanup(this.uppy.opts.id)
+      if (this.uppy.getState().recoveredState) {
```

Comment:
```suggestion
      if (this.uppy.getState().recoveredState != null) {
```

Reply:
Mikeal suggestions sounds good, we might go with strict inequality as I don't think we want `recoveredState` to ever be `undefined`.
```suggestion
      if (this.uppy.getState().recoveredState !== null) {
```

Reply:
why don't we want that? properties in JS are `undefined` by default before they are set, so I see it as likely that it could be `undefined`. Also a developer might some time in the future set recoveredState to `undefined`

Reply:
We only want to avoid the infinite recursion, not short-cutting the `cleanup` call when `recoveredState` happens to be `undefined` – not sure what kind of side effects it would have, but better safe than sorry.